internal/cli/plugin/plugin_asset.go

```diff
@@ -0,0 +1,253 @@
+
+	minParts := 2
+	if len(parts) < minParts {
+		return "", "", err
+	}
+	owner := parts[len(parts)-2]
+	name := parts[len(parts)-1]
+
+	if owner == "" || name == "" {
+		return "", "", err
+	}
+
+	return owner, name, nil
+}
+
+func parseReleaseVersion(arg string) (string, error) {
+	parts := strings.Split(arg, "@")
+
+	minParts := 2
+	if len(parts) < minParts {
+		return "", nil
+	}
+
+	version := parts[1]
+
+	if version == "" || version == "latest" {
+		return "", nil
+	}
+
+	parsedVersion, err := semver.NewVersion(version)
+
+	if err != nil {
+		return "", errors.New(`the specified version is invalid, it needs to follow the rules of Semantic Versioning`)
+	}
+
+	return parsedVersion.String(), nil
+}
+
+func extractTarGz(src string, dest string) error {
+	file, err := os.Open(src)
+	if err != nil {
+		return fmt.Errorf("failed to open .tar.gz file: %w", err)
+	}
+	defer file.Close()
+
+	gzipReader, err := gzip.NewReader(file)
+	if err != nil {
+		return fmt.Errorf("failed to create gzip reader: %w", err)
+	}
+	defer gzipReader.Close()
+
+	tarReader := tar.NewReader(gzipReader)
+
+	for {
+		header, err := tarReader.Next()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return errors.New("failed to read tar archive")
+		}
+
+		fileName := filepath.Clean(header.Name)
+		if strings.HasPrefix(fileName, "..") {
+			return fmt.Errorf("illegal file path for extracted plugin asset file: %s", fileName)
+		}
+
+		filePath := filepath.Join(dest, fileName)
+
+		switch header.Typeflag {
+		case tar.TypeDir:
+			if err := os.MkdirAll(filePath, os.FileMode(header.Mode)); err != nil {
+				return errors.New("failed to create directory while extracting plugin asset")
+			}
+		case tar.TypeReg:
+			outFile, err := os.Create(filePath)
+			if err != nil {
+				return errors.New("failed to create file while extracting plugin asset")
+			}
+			defer outFile.Close()
+			for {
+				_, err := io.CopyN(outFile, tarReader, 1024) //nolint:mnd // 1k each write to avoid compression bomb
+				if err != nil {
+					if err == io.EOF {
+						break
+					}
+					return errors.New("failed to copy file content while extracting plugin asset")
+				}
+			}
+
+		default:
+			return errors.New("failed to read tar archive")
+		}
+	}
+
+	return nil
+}
+
+func saveReadCloserToPluginAssetZipFile(rc io.ReadCloser) (string, error) {
+	defer rc.Close()
+
+	pluginsDefaultDirectory, err := plugin.GetDefaultPluginDirectory()
+	if err != nil {
+		return "", errors.New("could not find plugin directory")
+	}
+
+	pluginZipFilePath := filepath.Join(pluginsDefaultDirectory, "zippedPluginFiles.tar.gz")
+	pluginTarGzFile, err := os.Create(pluginZipFilePath)
+	if err != nil {
+		return "", errors.New("could not open plugins directory")
+	}
+	defer pluginTarGzFile.Close()
+
+	_, err = io.Copy(pluginTarGzFile, rc)
+	if err != nil {
+		return "", errors.New("failed to save asset to plugin directory: " + err.Error())
```

Comment:
Don't forget to clean up the file you made at line 222

Reply:
done